File: src/orden_compra.py
from datetime import date

class OrdenDeCompra:
    def __init__(self, id_orden: int, valor_total: float, fecha: date, estado: str):
        self.__id_orden = id_orden
        self.__valor_total = valor_total
        self.__fecha = fecha
        self.__estado = estado

    def confirmar(self):
        self.__estado = 'Confirmada'
        print("Su orden ha sido confirmada")

    def cancelar(self):
        self.__estado = 'Cancelada'
        print("Su orden ha sido cancelada")

# --- Getters ---
    def get_id_orden(self) ->int:
        return self.__id_orden

    def get_valor_total(self) ->float:
        return self.__valor_total

    def get_fecha(self) -> date:
        return self.__fecha

    def get_estado(self) -> str:
        return self.__estado

File: Televentas Java/Clases/OrdenCompra.java
package Clases;

import java.time.LocalDate;

public class OrdenCompra {

    private int idOrden;
    private double valorTotal;
    private LocalDate fecha;
    private String estado;

    public OrdenCompra(int idOrden, double valorTotal, LocalDate fecha, String estado) {
        this.idOrden = idOrden;
        this.valorTotal = valorTotal;
        this.fecha = fecha;
        this.estado = estado;
    }

    public void confirmar() {
        this.estado = "Confirmada";
        System.out.println("Su orden ha sido confirmada");
    }

    public void cancelar() {
        this.estado = "Cancelada";
        System.out.println("Su orden ha sido cancelada");
    }

    // --- Getters ---
    public int getIdOrden() { return idOrden; }
    public double getValorTotal() { return valorTotal; }
    public LocalDate getFecha() { return fecha; }
    public String getEstado() { return estado; }
}
